lib/bom.js

```diff
@@ -362,6 +366,55 @@ declare class MutationObserver {
     disconnect(): void;
 }
 
+declare class DOMRectReadOnly {
+    x: number,
+    y: number,
+    width: number,
+    height: number,
+    top: number,
+    right: number,
+    bottom: number,
+    left: number,
+    constructor(x: number, y: number, width: number, height: number): void,
+    fromRect(rectangle?: { x?: number, y?: number, width?: number, height?: number }): DOMRect,
+}
+
+declare class DOMRect extends DOMRectReadOnly {
+    constructor(x: number, y: number, width: number, height: number): void,
```

Comment:
These are all optionals too, according to https://drafts.fxtf.org/geometry/#DOMRect